pkg/auth/authenticator/password/ldappassword/ldap.go

```diff
@@ -15,41 +12,33 @@ import (
 
 	authapi "github.com/openshift/origin/pkg/auth/api"
 	"github.com/openshift/origin/pkg/auth/authenticator"
+	ldaputil "github.com/openshift/origin/pkg/auth/authenticator/password/ldappassword/util"
 )
 
 // Options contains configuration for an Authenticator instance
 type Options struct {
 	// URL is a parsed RFC 2255 URL
-	URL LDAPURL
-	// Insecure specifies if TLS is required for the connection. If true, either an ldap://... URL or StartTLS must be supported by the server
-	Insecure bool
-	// TLSConfig holds the TLS options. Only used when Insecure=false
-	TLSConfig *tls.Config
+	URL ldaputil.LDAPURL
+	// ServerConfig holds information about the LDAP server
+	ServerConfig ldaputil.LDAPServerConfig
 
 	// BindDN is the optional username to bind to for the search phase. If specified, BindPassword must also be set.
 	BindDN string
 	// BindPassword is the optional password to bind to for the search phase.
 	BindPassword string
 
-	// AttributeEmail is the optional list of LDAP attributes to use for the email address of the user identity.
-	// The first attribute with a non-empty value is used.
-	AttributeEmail []string
-	// AttributeName is the optional list of LDAP attributes to use for the display name of the user identity.
-	// The first attribute with a non-empty value is used.
-	AttributeName []string
-	// AttributePreferredUsername is the optional list of LDAP attributes to use for the preferred username of the user identity.
-	// The first attribute with a non-empty value is used. If not specified, the id determined by AttributeID is used as the preferred login.
-	AttributePreferredUsername []string
-	// AttributeID is the required list of LDAP attributes to use for the id address of the user identity.
-	// The first attribute with a non-empty value is used. If no attributes have values, login fails.
-	AttributeID []string
+	// IdentityAttributeMapping holds the lists of LDAP attributes to use for the email address, display name,
+	// preferred username and ID address of the user identity. The first attribute with a non-empty value is used
+	// for all but the latter identity field. If no LDAP attributes are given for the ID address, login fails.
+	IdentityAttributeMapping ldaputil.LDAPUserToIdentityMapping
 }
 
 // Authenticator validates username/passwords against an LDAP v3 server
 type Authenticator struct {
-	providerName string
-	options      Options
-	mapper       authapi.UserIdentityMapper
+	providerName    string
```

Comment:
No, this is the authenticator code for `ldappassword` - still needs the provider name.